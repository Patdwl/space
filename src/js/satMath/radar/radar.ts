export const radarMaxrng = (pW: number, aG: number, rcs: number, minSdB: number, fMhz: number): number => {
  // let powerInWatts = 325 * 1792;
  // let antennaGain = 2613000000;
  // let minimumDetectableSignaldB;
  let minSW = Math.pow(10, (minSdB - 30) / 10);
  // let frequencyMhz = 435;
  let fHz = (fMhz *= Math.pow(10, 6));
  let numer = pW * Math.pow(aG, 2) * rcs * Math.pow(3 * Math.pow(10, 8), 2);
  let denom = minSW * Math.pow(4 * Math.PI, 3) * Math.pow(fHz, 2);
  let rng = Math.sqrt(Math.sqrt(numer / denom));
  return rng;
};
export const radarMinSignal = (pW: number, aG: number, rcs: number, rng: number, fMhz: number): number => {
  // let powerInWatts = 325 * 1792;
  // let antennaGain = 2613000000;
  // let minimumDetectableSignaldB;
  // let frequencyMhz = 435;
  let fHz = (fMhz *= Math.pow(10, 6));
  let numer = pW * Math.pow(aG, 2) * rcs * Math.pow(3 * Math.pow(10, 8), 2);
  let denom = rng ** 4 * Math.pow(4 * Math.PI, 3) * Math.pow(fHz, 2);
  let minSW = numer / denom;
  let minSdB = Math.log10(minSW);
  return minSdB;
};
