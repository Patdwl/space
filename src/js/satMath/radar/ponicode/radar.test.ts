import * as radar from '@app/js/satMath/radar/radar';
// @ponicode
describe('radar.radarMinSignal', () => {
  test('0', () => {
    let result: any = radar.radarMinSignal(-100, 100, 0.0, 4.0, 10.23);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = radar.radarMinSignal(1, -100, 0.0, 5.0, 10.0);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = radar.radarMinSignal(-100, -5.48, 10.0, 4, 10.0);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = radar.radarMinSignal(0, 0, -0.5, 4, -29.45);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = radar.radarMinSignal(-5.48, -5.48, 1.0, 4.0, 0.5);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = radar.radarMinSignal(Infinity, Infinity, Infinity, Infinity, Infinity);
    expect(result).toMatchSnapshot();
  });
});

// @ponicode
describe('radar.radarMaxrng', () => {
  test('0', () => {
    let result: any = radar.radarMaxrng(-100, 100, 10.23, 31, -1.0);
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    let result: any = radar.radarMaxrng(100, -100, 10.23, 29, 1.0);
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    let result: any = radar.radarMaxrng(1, 0, -29.45, 31.0, 0.5);
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    let result: any = radar.radarMaxrng(-5.48, 1, 0.0, 29, 0.5);
    expect(result).toMatchSnapshot();
  });

  test('4', () => {
    let result: any = radar.radarMaxrng(100, 0, 10.0, 31, -0.5);
    expect(result).toMatchSnapshot();
  });

  test('5', () => {
    let result: any = radar.radarMaxrng(-Infinity, -Infinity, -Infinity, -Infinity, -Infinity);
    expect(result).toMatchSnapshot();
  });
});
