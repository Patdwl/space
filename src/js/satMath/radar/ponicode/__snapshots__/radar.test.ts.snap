// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radar.radarMaxrng 0 1`] = `NaN`;

exports[`radar.radarMaxrng 1 1`] = `155.4612437014322`;

exports[`radar.radarMaxrng 2 1`] = `-0`;

exports[`radar.radarMaxrng 3 1`] = `-0`;

exports[`radar.radarMaxrng 4 1`] = `0`;

exports[`radar.radarMaxrng 5 1`] = `NaN`;

exports[`radar.radarMinSignal 0 1`] = `-Infinity`;

exports[`radar.radarMinSignal 1 1`] = `-Infinity`;

exports[`radar.radarMinSignal 2 1`] = `NaN`;

exports[`radar.radarMinSignal 3 1`] = `-Infinity`;

exports[`radar.radarMinSignal 4 1`] = `NaN`;

exports[`radar.radarMinSignal 5 1`] = `NaN`;
